numpy/core/_internal.py

```diff
@@ -865,7 +868,10 @@ def npy_ctypes_check(cls):
     try:
         # ctypes class are new-style, so have an __mro__. This probably fails
         # for ctypes classes with multiple inheritance.
-        ctype_base = cls.__mro__[-2]
+        if IS_PYPY:
+            ctype_base = cls.__mro__[-3]
+        else:
+            ctype_base = cls.__mro__[-2]
```

Comment:
Can you add a brief comment here showing what the mro looks like in each case? Something like `# (..., _ctypes._CData, object)` for cpython

Reply:
In PyPy, `_ctypes` is implemented in pure python over an internal `_rawffi` module.
The mro is `# (..., <class '_ctypes.basics._CData'>, <class 'Bufferable'>, <class 'object'>)`, the [`Bufferable`](file:///home/matti/pypy_stuff/pypy/pypy/doc/_build/html/__pypy__-module.html#generally-available-functionality) class allows pure python support for PEP-3118 via an overridable `__buffer__` class method when calling `tp_as_buffer. bf_getbuffer` and using the `__pypy__.newmemoryview` function.

Adding the requested comment

Reply:
documentation for PyPy-specific `bufferable` and `__buffer__` is in the [`__pypy__`](http://doc.pypy.org/en/latest/__pypy__-module.html) module

Reply:
Nit: would probably abbreviate to `(..., _ctypes.basics._CData, Bufferable, object)` to match the CPython comment

Reply:
ok